application/TrajectoryCalculator/src/TrajectoryCalculator.cpp

```diff
@@ -0,0 +1,300 @@
+#include <include/TrajectoryCalculator.hpp>
+#include <numeric>
+
+#define MAX_COLLISIONS 5
+
+namespace TrajectoryCalculator
+{
+
+    void TrajectoryCalculator::setTableCorners(std::array<cv::Point, 4>& corners)
+    {
+        tableCorners = corners;
+    }
+
+    void TrajectoryCalculator::setBalls(std::vector<cv::Point>& balls)
+    {
+        ballLocations = balls;
+    }
+
+    void TrajectoryCalculator::setCue(std::array<cv::Point, 2>& cue)
+    {
+        cuePoints = cue;
+    }
+
+    void TrajectoryCalculator::setBallRadius(uint16_t radius)
+    {
+        ballRadius = radius;
+    }
+
+    void TrajectoryCalculator::setPocketRadius(uint16_t radius)
+    {
+        pocketRadius = radius;
+    }
+
+    std::vector<cv::Point> TrajectoryCalculator::getTrajectory()
+    {
+        std::vector<cv::Point> trajectory = {};
+
+        if (ballLocations.size() > 0)
+        {
+            float angle = ptToAngle(cuePoints[0], cuePoints[1]);
+            cv::Point fp = predictPoint(cuePoints[1], angle, tableCorners[2].x);
+            if (linePointDistance(cuePoints[0], fp, ballLocations[0]) <= ballRadius)
+            {
+                uint8_t collisions = 0;
+
+                cv::Point start = ballLocations[0];
+                trajectory.push_back(start);
+
+                std::vector<cv::Point> hitBalls = {};
+                uint8_t flag = 1;
+
+                while (collisions < MAX_COLLISIONS && flag == 1)
+                {
+                    trajectory.push_back(nextPoint(&start, &angle, &hitBalls, &flag));
+                    ++collisions;
+                }
+            }
+        }
+        
+        return trajectory;
+    }
+
+    cv::Point TrajectoryCalculator::nextPoint(cv::Point *start, float *angle, std::vector<cv::Point> *hitBalls, uint8_t *flag)

```

Comment:
pass by reference meaning function would look like nextPoint(cv::Point& start, float& angle, std::vector<cv::Point> &hitBalls, uint8_t &flag) achieves the same thing as pointers, but reduces accidental errors